### DISCUSSION TITLE
Code inspection warning

### DISCUSSION THREAD

**Discussion by ThomasCollignon** [ekvedaras/laravel-query-intellij#77] [Q&A] [answered]
Hello,
With the plugin activated I wonder if it is normal to still have this code inspection warning ?

![image](https://user-images.githubusercontent.com/16621630/145183395-c3a73d8d-4e02-43f4-90ff-1b9781175432.png)

Any advice on how to remove this warning without entirely deactivating the code inspection for methods that are not found ?

**Comment by ekvedaras** [comment]
Hey. You need either [Laravel Idea](https://laravel-idea.com) plugin (paid) or add [Laravel IDE helper](https://github.com/barryvdh/laravel-ide-helper) to your project and run 
```shell
php artisan ide-helper:generate
php artisan ide-helper:meta
php artisan ide-helper:models
```
which will generate some helper files so your IDE could see Eloquent methods.

Laravel Query plugin needs either of those to work otherwise, it cannot understand for which methods to trigger autocompletion.

**Reply by Hadjjo** [reply]
that wasnt mentioned in your plugin docs why ?

**Reply by ekvedaras** [reply]
Will add it 🙂

**Comment by ThomasCollignon** [comment, marked as answer]
The commands generate/metal/models have no effect for this particular inspection warning.
The command that actually makes this inspection warning disappear is
```
php artisan ide-helper:eloquent
```
